galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt

```diff
@@ -1413,23 +1414,141 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
-    TODO()
+  open fun load() {
+    // get select condition from first record in block
+    if (isMulti()) {
+      activeRecord = 0
+    }
+    callProtectedTrigger(VConstants.TRG_PREQRY)
+
+    // create database query
+    val columns = getSearchColumns_()
+    val table = getSearchTables_()
+    val condition = getSearchConditions_()
+    val orderBy = getSearchOrder_()
+
+    if (isMulti()) {
+      activeRecord = -1
+      activeField = null
+    }
+
+    // clear block: it will only hold the retrieved tuples
+    clear()
+
+    // get index of id field in BLOCK
+    val idfld: Int = getFieldIndex(idField)
+
+    // get index of id field in QUERY
+    var idqry = 0
+
+    for (i in 0 until idfld) {
+      if (fields[i].getColumnCount() > 0) {
+        idqry += 1
+      }
+    }
+
+    // open database query, fetch tuples
+    val query = table.slice(columns).select(condition).orderBy(*orderBy.toTypedArray())
+
+    fetchCount = 0
+    for (result in query) {
+      if (fetchCount < fetchSize) {
+        if (result[columns[1 + idqry]] == 0) {
+          continue
+        }
+        fetchBuffer[fetchCount] = result[columns[1 + idqry]] as Int
+        if (fetchCount >= bufferSize) {
+          fetchCount += 1
+        } else {
+          var i = 0
+          var j = 0
+
+          while (i < fields.size) {
+            if (fields[i].getColumnCount() > 0) {
+              fields[i].setQuery_(fetchCount, query, 1 + j)
+              j += 1
+            }
+            i++
+          }
+          setRecordFetched(fetchCount, true)
+          setRecordChanged(fetchCount, false)
+          setRecordDeleted(fetchCount, false)
+          try {
+            if (isMulti()) {
+              activeRecord = fetchCount
+            }
+            callProtectedTrigger(VConstants.TRG_POSTQRY)
+            if (isMulti()) {
+              activeRecord = -1
+            }
+            fetchCount += 1
+          } catch (e: VException) {
+            if (isMulti()) {
+              activeRecord = -1
+            }
+            if (e is VSkipRecordException) {
+              clearRecordImpl(fetchCount)
+            } else {
+              clear()
+              throw e
+            }
+          } catch (t: Throwable) {
+            t.printStackTrace()
+          }
+        }
+      }
+
+    }
+    fetchPosition = 0
+    // !!! REMOVE setActiveRecord(0);
+    if (!isMulti() && fetchCount == 0) {
+      throw VQueryNoRowException(MessageCode.getMessage("VIS-00022"))
+    } else if (!isMulti()) {
+      setMode(VConstants.MOD_UPDATE)
+    }
+    fireBlockChanged()
   }
 
   /**
    * Fetches record with given ID from database.
    * @exception VException      an exception may be raised by triggers
    */
-  fun fetchRecord(id: Int) {
-    TODO()
+  open fun fetchRecord(id: Int) {
+    val columns = getSearchColumns_()
+    val table = getSearchTables_()
+    val condition = Op.build { idColumn_ eq id  }
+    val query = table.slice(columns).select(condition)
```

Comment:
I think we might need to get other conditions from : VBlockDefaultOuterJoin.getFetchRecordCondition(fields)

Also move query inside the transaction block .